ibm/mas_devops/roles/cp4d_db2wh_backup/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+---
+# 1. Provide debug information to the user
+# -----------------------------------------------------------------------------
+- name: "DB2 Backup - Debug information"
+  debug:
+    msg:
+      - "CPD Namespace .......................... {{ cpd_meta_namespace_source }}"
+      - "DB2 Instance ID ........................ {{ db2wh_instance_id_source }}"
+      - "DB2 Backup Files Directory ............. {{ db2wh_backup_dir }}"
+
+# 2. Fail if no required keys are provided
+# -----------------------------------------------------------------------------
+- name: "Assert that cpd_meta_namespace_source is defined"
+  assert:
+    that:
+      - cpd_meta_namespace_source is defined
+      - cpd_meta_namespace_source != ""
+    fail_msg: "cpd_meta_namespace_source property is required"
+
+- name: "Assert that db2wh_instance_id_source is defined"
```

Comment:
Is this definitely an instance id versus instance name?  We use both vars in other db2 roles and just want to check we are being consistent with our naming conventions here :) 

In cpd4 instance id == instance name and we determine the id from the name input (https://github.com/ibm-mas/ansible-devops/blob/master/ibm/mas_devops/roles/cp4d_db2wh/tasks/cpd40/provision-instance.yml#L171), but in cpd3 they are different because the API is being used to create the database and again we use the name to determine the instance id, but rely on the additional configmap created during deploy.

if this needs to be instance id, I think we should have the same logic we have in cp4d_db2wh in here to convert from db2wh_instance_name to the ID that we need

Reply:
Here we just need something to lookup for the `db2wh` main pod, so it would be something as `db2wh-1637258370280916` (in case of db2wh created via api, cpd v3.5) or `db2wh-iot` (in case of db2wh created via db2ucluster, cpd v4)
Regardless for the case, do we need to add this conversion here? I'm thinking if we need to use this playbook in conjuction with another, can't we pass the `db2wh_instance_id_source` with the expected converted output from previous defined by other playbooks?
Just not sure if worths adding such cpd_version conversion for a role/playbook that works the same way regardless of cpd_version.

Reply:
fixed to be same as in `cp4d_db2wh` and `cp4d_db2wh_manage_hack` as you suggested.